ca/ca.go

```diff
@@ -154,6 +155,9 @@ func (ca *CA) Init(cfg *config.Config) (*CA, error) {
 		opts = append(opts, authority.WithDatabase(ca.opts.database))
 	}
 
+	webhookTransport := getWebhookTransport()
+	opts = append(opts, authority.WithWebhookClient(&http.Client{Transport: webhookTransport}))
```

Comment:
The client must not follow redirects. For hosted, we must ensure it never resolves to a private network range. We may need to think about how to do this when deployed in own infrastructure, because generally we can probably allow resolving to private network ranges, but there may be cases where it's not OK. We may also need to think about malicious servers exhausting resources.

Reply:
798652a08ce9882c321c16f5d168e5c94f71a7d7
With this commit you can set a policy in ca.json such as
```
"webhookClient": {
  "allow": {
    "public": true,
    "hostnames": ["customer.smallstep.svc.cluster.local"],
  }
}
```
The policy will be checked before every dial so I think it's safe to follow redirects.

Do you have anything in mind for protecting against malicious servers other than setting a timeout?

Reply:
Was indeed thinking about a timeout. Another is to try to detect a really slow server response, like what https://github.com/skeeto/endlessh does to SSH clients, but maybe that's a stretch.

Reply:
This client policy should not be necessary, the OS and TLS address most concerts. If I want to restrict an IP, I will use a firewall instead of this.

Reply:
a78aa3c2